lib/goddess/custom_query_container.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+module Goddess
+  class CustomQueryContainer < Valkyrie::Persistence::CustomQueryContainer
+    ##
+    # @note What do we do when we have an empty array returned in the first query service?
+    def method_missing(method_name, *args, **opts)
+      # As we iterate through the services, we need to know if any of them responded to the given
+      # method_name.
+      service_responds_to = false
+      returning_value = nil
+
+      query_service.services.each do |service|
+        next unless service.custom_queries.respond_to?(method_name)
+        service_responds_to = true
+        returning_value = service.custom_queries.send(method_name, *args, **opts)
+        # Note, an empty array is "true" in this case.  Should it be?
+        break if returning_value
+
+        # If we don't find the resource in the first service, we should try again in the second
+        # service, and so forth
+      rescue Valkyrie::Persistence::ObjectNotFoundError
+        next
+      end
+
+      # None of the services responded to the method_name
+      return super unless service_responds_to
+
+      # At least one of the services had a valid query result; though it could be an empty array.
+      # Is that okay?  What if we would have had a non-empty array in a later service?
+      return returning_value if returning_value
+
+      # We didn't find anything so we need to re-raise the like error
+      raise Valkyrie::Persistence::ObjectNotFoundError, "#{self.class}#method_missing with method_name: #{method_name.inspect}, args: #{args.inspect}, opts: #{opts.inspect}"
```

Comment:
if you create a variable (`e`) on line 14, then set it if need by on line 21, then this can simply be `raise e` and will walk the backtrace up the line. not sure if thats more info that you're getting now or less and I'd go with whatever gives us more actionable info.